File: app/src/main/java/com/nilhcem/droidconde/data/app/model/Room.java
package com.nilhcem.droidconde.data.app.model;

import android.support.annotation.NonNull;

public enum Room {

    NONE(0, ""),
    STAGE_1(1, "Stage1"),
    STAGE_2(2, "Stage2"),
    STAGE_3(3, "Stage3"),
    STAGE_4(4, "Stage4"),
    WORKSPACE(5, "Workspace");

    public final int id;
    public final String label;

    Room(int id, String label) {
        this.id = id;
        this.label = label;
    }

    public static Room getFromId(int id) {
        for (Room room : Room.values()) {
            if (room.id == id) {
                return room;
            }
        }
        return NONE;
    }

    public static Room getFromLabel(@NonNull String label) {
        for (Room room : Room.values()) {
            if (label.equals(room.label)) {
                return room;
            }
        }
        return NONE;
    }
}

File: scraper/src/main/kotlin/com/nilhcem/droidconde/scraper/model/Room.kt
package com.nilhcem.droidconde.scraper.model

enum class Room(val id: Int, val roomName: String) {

    NONE(0, ""),
    STAGE_1(1, "Stage 1"),
    STAGE_2(2, "Stage 2"),
    STAGE_3(3, "Stage 3"),
    STAGE_4(4, "Stage 4"),
    WORKSPACE(5, "Workspace");

    companion object {
        fun getRoomId(name: String) = Room.values().filter { name == it.roomName }.map { it.id }.getOrElse(0, { NONE.id })
    }
}
